### PR TITLE
Support for null safety

### PR DESCRIPTION
**Pull request by yanivshaked**
Null safety support, version changed to 1.0.0.

### CODE REVIEW

File: example/lib/main.dart [hunk 1]
```diff
@@ -52,17 +52,17 @@ class _MyAppState extends State<MyApp> {
   }
 
   Future<void> getPermissionStatus() async {
-    bool permissionStatus = false;
+    bool? permissionStatus = false;
     try {
-      permissionStatus = await PermissionHandler.permissionsGranted;
+      permissionStatus = await (PermissionHandler.permissionsGranted as FutureOr<bool>);
```
Comment (TryingOutSomething) [inline example/lib/main.dart]: Could you remove `as FutureOr<bool>` from this line? It causes the PermissionHandler to not work as intended. The console logs the following line while launching the app from your pull request:
```
type 'Future<bool?>' is not a subtype of type 'FutureOr<bool>' in type cast
```

**Comment by TryingOutSomething** [comment]
Hi @yanivshaked, I have reviewed your pull request. Do you mind making a small change to the pull request before I merge to the main branch and integrate it into the plugin?

File: example/lib/main.dart (hunk 1 above)
Comment (yanivshaked) [reply example/lib/main.dart]: Removed, thanks!

**Comment by yanivshaked** [comment]
I have committed the requested change, thanks!

**Comment by TryingOutSomething** [comment]
Thanks for making the change!

**Pull request merged**